modules/core/shared/src/main/scala/gem/math/ObservingNight.scala

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import gem.enum.Site
+
+import cats._
+import cats.data.Validated
+import cats.effect.IO
+import cats.implicits._
+
+import java.time._
+import java.time.format.DateTimeFormatter
+
+import scala.math.Ordering.Implicits._
+
+
+/** An observing night is defined as the period of time from 14:00 on one day
+  * until 14:00 on the following day.  In a timezone that honors daylight saving,
+  * it is sometimes onger and sometimes shorter than a period of 24 hours.  This
+  * is also true of days which contain leap seconds.
+  */
+final case class ObservingNight private (start: Instant, end: Instant, site: Site) extends Night {
```

Comment:
Did you intend `sealed abstract` here? The `private` constructor doesn't hide the factory methods and `copy`.

Reply:
D'oh! I knew that, and yet screwed it up.  Thanks.